### PR TITLE
Load model callbacks

### LINKED ISSUE #11279: load_model does not call Callbacks when setting the model
The load_model path currently does not call the Callback's set_model method. Since the load_model method loads and sets model and optimizer weights this poses a problem for Callbacks that update the model at set_model time since graph modifications are not allowed after the model is run by a session.

With the TensorFlow backend the model and optimizer weights are set by load_model calling `K.batch_set_value` which puts the model in a session.

If the callback set_model is called after the load_model is complete it leads to errors like this: 

`tensorflow/c/c_api.cc:686] Operation '{name:'training/Adam/gradients/AddN_755' id:5241 op device:{} def:{training/Adam/gradients/AddN_755 = AddN[N=2, T=DT_FLOAT, _class=["loc:@up_sampling3d_2/concat_2"]](lms/swapin_1139:0, training/Adam/gradients/up_sampling3d_2/concat_2_grad/split:31)}}' was changed by updating input tensor after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.
`

I have tested a solution on tf.keras at the 2.2.0 level with TensorFlow 1.10 where the callbacks are passed in as a keyword arg to load_model and the callbacks' set_model is called after the model is loaded and build but before the weights are set. I will use this issue to put a PR up for that solution.

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ X ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

### PR DESCRIPTION
**Pull request by smatzek**
### Summary
Call set_model on callbacks during load_model

The load_model function builds the model from the saved object, and
implicitly sets the model in the backed when the weights are set.

Callbacks that require set_model to be called before the model is
set on the backend can fail to operate on loaded models as a result.

This commit adds the ability to pass callbacks to the load_model function.### Summary

### Related Issues
Fixes #11279
### PR Overview

- [ Y ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ N ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ Y ] This PR is backwards compatible [y/n]
- [ Y ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Hello and thanks for the PR. For any change in the API, a google doc explaining the motivation should first be done and reviewed. Only after that can someone make a PR implementing this API change. See https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md#pull-requests for more info on how to do that. I'll close the PR for now. It'll be reopened if/when the google doc get approved (add a message in this thread and I'll reopen). 

Thank you for having invested your time into this.
